zxlive/dialogs.py

```diff
@@ -210,6 +210,7 @@ def save_diagram_dialog(graph: GraphT, parent: QWidget) -> Optional[tuple[str, F
     file_path, selected_format = file_path_and_format
 
     if selected_format in (FileFormat.QGraph, FileFormat.Json):
+        graph.auto_detect_io()
```

Comment:
graph.auto_detect_io() can sometimes fail. Can you put it inside a try block so that the save works even when detecting io failed?
```python
try:
    graph.auto_detect_io()
except TypeError:
    pass
```

Reply:
Ah I see, I have updated the PR :-)